### ISSUE TITLE
ParakeetForCTC outputs only blank tokens - empty transcription on all backends (v4.2.0)

### ISSUE THREAD

**Issue by madforstrength** [huggingface/transformers.js#1726]
### System Info

- `@huggingface/transformers` **4.2.0** (latest)
- Reproduced on **two independent backends**:
  - Node.js v25.8.2, Windows 11, CPU, `dtype: 'q8'`
  - Chrome extension (offscreen document worker), WebGPU, `dtype: 'fp16'`
- Model: [`onnx-community/parakeet-ctc-0.6b-ONNX`](https://huggingface.co/onnx-community/parakeet-ctc-0.6b-ONNX)

### Description

`pipeline('automatic-speech-recognition', 'onnx-community/parakeet-ctc-0.6b-ONNX')` returns an **empty string for clear speech audio**, on every backend/dtype combination we tried. There is no error — just `{"text": ""}`.

A console warning is also emitted during load:

```
Unknown tokenizer class "ParakeetTokenizer", attempting to construct from base class.
```

but the tokenizer turns out not to be the culprit (see investigation below).

### Reproduction 1 — pipeline returns empty text

```js
import { pipeline } from '@huggingface/transformers';

// jfk.wav: 16 kHz mono 16-bit PCM speech sample
const res = await fetch('https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/jfk.wav');
const buf = Buffer.from(await res.arrayBuffer());
// ...decode WAV data chunk to Float32Array `audio` (s16le / 32768)...

const asr = await pipeline('automatic-speech-recognition', 'onnx-community/parakeet-ctc-0.6b-ONNX', { dtype: 'q8' });
const out = await asr(audio);
console.log(JSON.stringify(out)); // -> {"text":""}
```

Observed: `{"text":""}` after ~11s of inference on 60s of clear speech (so the model *ran*, it just produced nothing).

### Reproduction 2 — raw logits are 100% blank tokens

To isolate decode vs. model, we bypassed the pipeline:

```js
import { AutoProcessor, AutoModel, AutoTokenizer } from '@huggingface/transformers';

const ID = 'onnx-community/parakeet-ctc-0.6b-ONNX';
const processor = await AutoProcessor.from_pretrained(ID);
const model = await AutoModel.from_pretrained(ID, { dtype: 'q8' });

const inputs = await processor(audio);            // keys: input_features, attention_mask
const { logits } = await model(inputs);           // dims: [1, 88, 1025] for 7s of audio

// per-frame argmax
const [, frames, vocab] = logits.dims;
const ids = [];
for (let t = 0; t < frames; t++) {
  let best = 0, bestV = -Infinity;
  for (let v = 0; v < vocab; v++) {
    const val = logits.data[t * vocab + v];
    if (val > bestV) { bestV = val; best = v; }
  }
  ids.push(best);
}
```

Result: **all 88 frames argmax to id 1024** (the CTC blank, last index of the 1025-token vocab) for 7 seconds of clear speech. CTC-collapsing therefore yields zero tokens, and the empty pipeline output follows trivially.

### Interpretation

Since the model predicts blank at literally every frame — identically under Node/CPU/q8 and browser/WebGPU/fp16 — the problem appears to be upstream of decoding: either `ParakeetFeatureExtractor` produces features the exported model can't consume, or the ONNX export itself is faulty. The `ParakeetTokenizer` fallback warning is a red herring for this particular failure (the tokenizer never receives any tokens to decode).

For comparison, Whisper and `cohere-transcribe` models transcribe the same audio correctly in the same environments, so the audio path and harness are fine.

### Expected behavior

A transcription of the speech sample (e.g. the JFK quote), or at minimum a loud failure instead of a silent empty string.

Possibly related (but distinct): #1310 is a request for Parakeet **TDT** support; this issue is about the already-supported **CTC** variant producing degenerate output.

Happy to run further diagnostics if useful — we hit this building an open-source on-device dictation extension and would love to ship Parakeet as an engine.

**Comment by hogeheer499-commits** [comment]
I reproduced the empty output, but it comes from the WAV decoding path rather than Parakeet or the ONNX export. The linked `jfk.wav` is **44.1 kHz, stereo, 16-bit PCM**, not 16 kHz mono:

```text
{ audioFormat: 1, channels: 2, sampleRate: 44100, bits: 16 }
```

If its interleaved PCM samples are only divided by 32768 and passed directly as a `Float32Array`, Transformers.js has no container metadata and treats those samples as 16 kHz mono. With that input I get the reported result on current `main` and on `@huggingface/transformers@4.2.0`:

```json
{"text":""}
```

After downmixing to mono and resampling to 16 kHz first, the same model/dtype/package returns:

```json
{"text":"and so my fellow americans ask not what your country can do for you ask what you can do for your country"}
```

So the tokenizer warning is indeed unrelated, but the model features and export appear healthy. In Node, the decoded array needs to be explicitly downmixed/resampled before it is passed to the pipeline (the pipeline cannot infer the original rate/channels from a bare `Float32Array`).

**Comment by madforstrength** [comment]
You're absolutely right — thank you for the careful analysis. `jfk.wav` is 44.1 kHz stereo, and my test helper decoded it assuming 16 kHz mono, so the model was fed noise. The all-blank logits were Parakeet **correctly** reporting non-speech, not a broken export.

After downmixing and resampling to 16 kHz, `@huggingface/transformers@4.2.0` with `onnx-community/parakeet-ctc-0.6b-ONNX` (q8, CPU) returns:

```json
{"text":"and so my fellow americans ask not what your country can do for"}
```

(first 7 s slice of the clip — 72 of 88 frames non-blank, exactly as expected).

The in-browser empty transcripts that originally prompted this report are also explained by problems on my side: an OS-level silent microphone device at the time of testing, plus an earlier attempt using 4-bit encoder quantization on WebGPU. The `ParakeetTokenizer` warning is cosmetic, as you noted — the base-class fallback decodes CTC output fine.

Model, export, feature extraction, and decoding are all healthy. Closing — apologies for the noise, and thanks again.
